packages/core/src/fhirpath/parse.test.ts

```diff
@@ -601,4 +601,24 @@ describe('FHIRPath parser', () => {
     expect(evalFhirPath(expr, { concept, system })).toEqual([true]);
     expect(evalFhirPath(expr, { concept, filter, system })).toEqual([true]);
   });
+
+  test('where and', () => {
+    const expr = "identifier.where(system='http://example.com' and value='123').exists()";
```

Comment:
This test already worked out of the box.  This was my starting point for investigating #3753 - I'm leaving it in, because we did not have any tests for `where(x and y)`.